### PR TITLE
Add support for explicit protocol versioning

### LINKED ISSUE #14: Version negotiation?
Do we want to have some sort of version negotiation take place when the protocol is first connected? If we think we'll ever want this, specifying it early on will mean all implementations will support it at least well enough to say "no I don't support your protocol version". On the other hand, I expect most implementation distributions will be tightly-coupled enough that the embedder will always know which version of the compiler it's shipping (at least to within a semantic version range).

@strychnide What do you think?

### PR DESCRIPTION
**Pull request by nex3**
This doesn't add actual version negotiation, because it's expected
that embedded hots will tightly control which compiler versions they
install.

Partially addresses #37
Closes #14

### CODE REVIEW

**Comment by nex3** [comment]
cc @johnfairh

File: embedded_sass.proto [hunk 1]
```diff
@@ -8,8 +8,18 @@ package sass.embedded_protocol;
 
 // The wrapper type for all messages sent from the host to the compiler. This
 // provides a `oneof` that makes it possible to determine the type of each
-// inbound message.
+// inbound message.currently
```
Comment (johnfairh) [inline embedded_sass.proto:11]: (typo)

File: embedded_sass.proto [hunk 2]
```diff
@@ -231,13 +241,32 @@ message InboundMessage {
...
     FunctionCallResponse functionCallResponse = 6;
+    VersionRequest versionRequest = 7;
   }
 }
 
 // The wrapper type for all messages sent from the compiler to the host. This
 // provides a `oneof` that makes it possible to determine the type of each
 // outbound message.
 message OutboundMessage {
+  // A response that contains the version of the embedded host.
```
Comment (johnfairh) [inline embedded_sass.proto:252]:
```suggestion
  // A response that contains the version of the embedded compiler.
```
?

File: embedded_sass.proto [hunk 3]
```diff
@@ -231,13 +241,32 @@ message InboundMessage {
...
+  message VersionResponse {
+    // The version of the embedded protocol, in semver format.
+    string protocol_version = 1;
+
+    // The version of the embedded compiler package. This has no guaranteed
+    // format, although compilers are encouraged to use semver.
+    string compiler_version = 3;
```
Comment (Awjin) [inline embedded_sass.proto:259]:
```suggestion
    string compiler_version = 2;
```

File: embedded_sass.proto [hunk 4]
```diff
@@ -231,13 +241,32 @@ message InboundMessage {
...
+
+    // The version of the Sass implementation that the embedded compiler wraps.
+    // This has no guaranteed format, although Sass implementations are
+    // encouraged to use semver.
+    string implementation_version = 4;
```
Comment (Awjin) [inline embedded_sass.proto:264]:
```suggestion
    string implementation_version = 3;
```

**Review by Awjin — APPROVED** [review]
LGTM pending minor fixes

**Pull request merged**